src/layer/text.js

```diff
@@ -404,21 +414,131 @@ var Text = function(parentEl) {
             for (var i=0; i<count; i++) {
                 parent.appendChild(this.$tabStrings[" "].cloneNode(true));
             }
+            this.$highlightIndentGuide();
             return value.substr(cols);
         } else if (value[0] == "\t") {
             for (var i=0; i<cols; i++) {
                 parent.appendChild(this.$tabStrings["\t"].cloneNode(true));
             }
+            this.$highlightIndentGuide();
             return value.substr(cols);
         }
+        this.$highlightIndentGuide();
         return value;
     };
 
+    this.$highlightIndentGuide = function () {
+        if (!this.highlightIndentGuides) return;
+
+        this.$highlightIndentGuideMarker = {
+            indentLevel: undefined,
+            start: undefined,
+            end: undefined,
+            dir: undefined
+        };
+        var lines = this.session.doc.$lines;
+        var cursor = this.session.selection.getCursor();
+        var initialIndent = /^\s*/.exec(this.session.doc.getLine(cursor.row))[0].length;
```

Comment:
what if there are no spaces in the line? it seems like this line would throw exception

Reply:
It would just return 0 in that case, as i am using `*` quantifier